src/java.base/share/classes/java/lang/module/ModuleReader.java

```diff
@@ -213,9 +213,17 @@ default void release(ByteBuffer bb) {
      * when using the stream to list the module contents and access is denied
      * by the security manager. </p>
      *
+     * <p> The returned stream contains references to one or more open directories
+     * in the module. The directories are closed by closing the stream. </p>
```

Comment:
This mostly looks good - I agree with adding the note here. Should the wording be relaxed a little - replace "contains references" with "may contain references" - since the stream will not always contain references to open directories - it depends on the type of reader. The end result / guidance is the same - use in a try-with-resources block - I'm just suggesting a slight relaxing of the wording.

Reply:
"may contain references" is good.   I agree that the wordings should be relaxed a little.  Similar for the apiNote.